lib/smart_proxy_dns_route53/dns_route53_main.rb

```diff
@@ -53,9 +53,14 @@ def resolver
       @resolver ||= Resolv::DNS.new
     end
 
-    def get_zone(fqdn)
-      domain = fqdn.split('.', 2).last
-      conn.get_zones(domain)[0]
+    def get_zone(name)
```

Comment:
Can be greatly simplified when https://github.com/pcorliss/ruby_route_53/pull/43 is merged and in a release.